### PR TITLE
Add wallet check overlay

### LINKED ISSUE #1080: Support full screen web browser
When browsing in full screen, Firefox does not visibly open a popup with Metamask. The user has to click on the Metamask icon to actually approve the transaction (the Metamask icon will actually show a very small number overlay to indicate that something is expected but that's fairly hard to see!)

A solution to that issue is to actually show something like "Check your web3 provider" when a transaction is being initiated so that the user knows that something is expected from them.
We could either do that globally across the app (by applying a window wide lightbox which says "check your web3Provider" or something custom for each transaction but I think the first approach would be easier for now.

What do you think @smombartz ?

### PR DESCRIPTION
**Pull request by cnasc**
# Description
This PR fixes #1080. It adds a `WalletCheckOverlay` component to the top-level of the app. When the `walletStatus` in the state is set to `{ waiting: true }`, it renders an overlay that instructs the user to interact with their wallet to complete the transaction. 

**This isn't quite ready to go yet, but I would appreciate some review.**

- I think the appearance of this component on mobile is still an open question.
- At the moment the `ModalOverlay` file is named as such in case we want to create other descendants of the overlay. Maybe it would be a good idea to rename it to only have the specific focus of blocking the page for wallet interaction.

image

# Checklist:

- [x] 1 PR, 1 purpose: my Pull Request applies to a single purpose
  - [ ] This PR only contains configuration changes (package.json, etc.)
  - [x] This PR only contains code changes (if configuration changes are required, do a separate PR first, then re-base)
- [x] My code follows the style guidelines of this project, including naming conventions
- [x] I have commented my code, particularly in hard-to-understand areas
- [ ] I have added tests that prove my fix is effective or that my feature works
- [x] If my code adds or changes components, I have written corresponding stories with Storybook
- [x] I have performed a self-review of my own code
- [x] If my code involves visual changes, I am adding applicable screenshots to this thread

PS: [Read how to write the perfect pull request](https://blog.github.com/2015-01-21-how-to-write-the-perfect-pull-request/)

### CODE REVIEW

File: unlock-app/src/components/interface/ModalOverlay.js [hunk 1]
```diff
@@ -0,0 +1,91 @@
...
+  const {
+    walletStatus: { waiting },
+  } = state
+  return { waiting }
+}
+
+const Dismiss = styled.button`
+  height: 24px;
+  font-size: 20px;
+  font-family: Roboto;
```
Comment (cellog) [inline unlock-app/src/components/interface/ModalOverlay.js]: provide a fallback font?

File: unlock-app/src/components/interface/ModalOverlay.js [hunk 2]
```diff
@@ -0,0 +1,91 @@
...
+  background: rgba(0, 0, 0, 0.5);
+  position: fixed;
+  height: 100%;
+  width: 100%;
+  left: 0;
+  top: 0;
+  display: flex;
+  flex-direction: column;
+  justify-content: center;
+  align-items: center;
```
Comment (cellog) [inline unlock-app/src/components/interface/ModalOverlay.js]: add `z-index: var(--alwaysontop)`?

File: unlock-app/src/components/interface/ModalOverlay.js [hunk 3]
```diff
@@ -0,0 +1,91 @@
+import React from 'react'
+import { connect } from 'react-redux'
+import styled from 'styled-components'
+import PropTypes from 'prop-types'
+import { dismissWalletCheck } from '../../actions/walletStatus'
+
+const ModalOverlay = ({ children }) => (
```
Comment (cellog) [inline unlock-app/src/components/interface/ModalOverlay.js]: perhaps rename to `FullScreenModal`

Comment (cnasc) [reply unlock-app/src/components/interface/ModalOverlay.js]: Rename the file, right?

Comment (cellog) [reply unlock-app/src/components/interface/ModalOverlay.js]: yes. Since it contains multiple components, I might go for `FullScreenModals.js`. Up to you.

**Pull request merged**